src/stactools/core/utils/convert.py

```diff
@@ -16,25 +17,102 @@
 """The default profile to use when writing Cloud-Optimized GeoTIFFs (COGs)."""
 
 
-def cogify(infile: str, outfile: str, profile: Optional[Dict[str, Any]] = None) -> None:
-    """Creates a Cloud-Optimized GeoTIFF (COG) from a GDAL-readable file.
+def check_gdal_driver() -> None:
+    if not utils.gdal_driver_is_enabled("COG"):
+        raise DriverRegistrationError(
+            "GDAL's COG driver is not enabled, make sure you're using GDAL >= 3.1"
+        )
+
+
+def cogify(
+    infile: str,
+    outfile: str,
+    band: Optional[int] = None,
+    profile: Optional[Dict[str, Any]] = None,
+) -> None:
+    """Creates a Cloud-Optimized GeoTIFF (COG) from a GDAL-readable file
+    without subdatasets (non-HDF files). A band number can optionally be
+    provided to extract a single band from a multiband file.
 
     Uses :py:meth:`rasterio.shutil.copy`.
 
     Args:
         infile (str): The input file.
         outfile (str): The output COG to be written.
+        band (Optional[int]): The band number in the input file to extract.
         profile (Optional[dict[str, Any]]):
             An optional profile to use on the
             output file. If not provided,
             :py:const:`stactools.core.utils.convert.DEFAULT_PROFILE` will be
             used.
     """
-    if not utils.gdal_driver_is_enabled("COG"):
-        raise DriverRegistrationError(
-            "GDAL's COG driver is not enabled, make sure you're using GDAL >= 3.1"
-        )
-    destination_profile = DEFAULT_PROFILE.copy()
+    check_gdal_driver()
+
+    src = rasterio.open(infile)
+    dest_profile = DEFAULT_PROFILE.copy()
+    dest_profile.update(
+        {
+            "dtype": rasterio.uint8,
+            "width": src.width,
+            "height": src.height,
+            "crs": src.crs,
+            "transform": src.transform,
+        }
+    )
+
     if profile:
-        destination_profile.update(profile)
+        dest_profile.update(profile)
+
+    # If a band number was provided, create a single-band COG
+    if band:
+        single_band = src.read(band)
+        dest_profile.update({"count": 1})
+        with rasterio.open(outfile, "w", **dest_profile) as dest:
+            dest.write(single_band, 1)
+    # If no band numbers were provided, create a multi-band COG
+    else:
+        dest_profile.update({"count": src.count})
+        rasterio.shutil.copy(infile, outfile, **dest_profile)
+
+
+def cogify_subdataset(infile: str, outfile: str) -> None:
+    """Exports a Cloud-Optimized GeoTIFF (COG) from a subdataset within an HDF file.
+    Uses :py:meth:`rasterio.shutil.copy`.
+    Args:
+        infile (str): The input file.
+        outfile (str): The output COG to be written.
+    """
+    check_gdal_driver()
+    destination_profile = DEFAULT_PROFILE.copy()
     rasterio.shutil.copy(infile, outfile, **destination_profile)
+
+
+def list_subdataset(infile: str, outdir: str) -> Tuple[List[str], List[str]]:
```

Comment:
```suggestion
def cogify_subdatasets(infile: str, outdir: str) -> Tuple[List[str], List[str]]:
```

Can we also add an optional argument that is a list of subdataset names that can be used to subset the output datasets? E.g.:

```python
def cogify_subdatasets(infile: str, outdir: str, subdataset_names: Optional[List[str]] = None) -> ...
```